src/node_buffer.cc

```diff
@@ -142,21 +138,14 @@ CallbackInfo::~CallbackInfo() {
 
 
 void CallbackInfo::WeakCallback(
-    const WeakCallbackData<Object, CallbackInfo>& data) {
-  data.GetParameter()->WeakCallback(data.GetIsolate(), data.GetValue());
+    const WeakCallbackInfo<CallbackInfo>& data) {
+  data.GetParameter()->WeakCallback(data.GetIsolate(),
+      static_cast<char*>(data.GetInternalField(kBufferInternalFieldIndex)));
```

Comment:
A not hugely important style nit: it's a little more idiomatic to write this as:
```c++
data.GetParameter()->WeakCallback(
    data.GetIsolate(),
    static_cast<char*>(data.GetInternalField(kBufferInternalFieldIndex)));
```